### PR TITLE
Fix failing Py37 BQ file loads test

### LINKED ISSUE #23264: [Bug]: Py37 PostCommit Failure test apache_beam.io.gcp.bigquery_file_loads_test.BigQueryFileLoadsIT.test_bqfl_streaming
### What happened?

Has been failing since Sep 14, from https://ci-beam.apache.org/job/beam_PostCommit_Python37/5719/. The test was skipped in Py38 PostCommit and only run on Py37 PostCommit.

Error message:

```
Error Message
google.api_core.exceptions.NotFound: 404 Not found: Table apache-beam-testing:python_bq_file_loads_1663183712232.output_table_ints was not found in location US

Location: US
Job ID: baf2a9a6-ecd8-49b7-a6ef-9965c7fd0cd9
```

Caused by #23012

### Issue Priority

Priority: 1

### Issue Component

Component: test-failures

### PR DESCRIPTION
**Pull request by ahmedabu98**
Fixes #23264

In #23012, testing with `on_success_matcher` is switched out for `hamcrest_assert(p, bq_matcher)` because it is less noisy and more accurate.

However, [PostCommits test with `TestDirectRunner`](https://github.com/apache/beam/blob/468e02b7c1f41f25543edbafb84becb07974eebe/sdks/python/test-suites/direct/common.gradle#L169-L177). The problem is that `TestDirectRunner` does not run the pipeline without the `on_success_matcher` argument. 

Error results because `hamcrest_assert` queries a table that the pipeline is supposed to create. Pipeline doesn't run so results in 404 error.

### CODE REVIEW

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Current test failure: https://ci-beam.apache.org/job/beam_PostCommit_Python37_PR/432/testReport/apache_beam.io.gcp.bigquery_file_loads_test/BigQueryFileLoadsIT/test_bqfl_streaming/

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python PreCommit

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
R: @Abacn 
postcommits failing for other reasons

**Comment by Abacn** [comment]
> R: @Abacn postcommits failing for other reasons

Python PostCommit timeout possibly related to https://github.com/apache/beam/pull/23334#issuecomment-1260015248

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by Abacn** [comment]
Changes only involve file_load_it_test and the other failed bigquery it tests (json_it) should be irrelevant.

Could you please explain a little of the fix---does it mean #23012 introduced some breaking change, or just the failed test did not configure correctly at the beginning?

File: sdks/python/test-suites/direct/common.gradle [hunk 1]
```diff
@@ -162,6 +162,8 @@ tasks.register("directRunnerIT") {
         "apache_beam/io/gcp/bigquery_test.py::BigQueryStreamingInsertTransformIntegrationTests::test_multiple_destinations_transform",
         "apache_beam/io/gcp/bigquery_test.py::PubSubBigQueryIT",
         "apache_beam/io/gcp/bigquery_file_loads_test.py::BigQueryFileLoadsIT::test_bqfl_streaming",
+        "apache_beam/io/gcp/bigquery_file_loads_test.py::BigQueryFileLoadsIT::test_bqfl_streaming_with_dynamic_destinations",
+        "apache_beam/io/gcp/bigquery_file_loads_test.py::BigQueryFileLoadsIT::test_one_job_fails_all_jobs_fail",
```
Comment (Abacn) [inline sdks/python/test-suites/direct/common.gradle:166]: If I understood correctly this line should refer to `test_bqfl_streaming_with_copy_jobs` (see skipped test in https://ci-beam.apache.org/job/beam_PostCommit_Python37_PR/444/testReport/apache_beam.io.gcp.bigquery_file_loads_test/BigQueryFileLoadsIT/)

Comment (ahmedabu98) [reply sdks/python/test-suites/direct/common.gradle]: Fixed. Good catch

**Comment by ahmedabu98** [comment]
> Could you please explain a little of the fix---does it mean #23012 introduced some breaking change, or just the failed test did not configure correctly at the beginning?

No breaking change from #23012 AFAIK. The test here was failing because in #23012 the test was changed to use [`hamcrest_assert()` instead of `on_success_matcher`](https://github.com/apache/beam/pull/23012/files#diff-915e1c8ed9209559ad337ac1a333648914b68c612ae99616a3bfe7f06fad1cefR939). For reasons unknown to me, `on_success_matcher` is needed when running the test using `./run_integration_test.sh` (which is what Postcommits are doing), otherwise the pipeline in the test never runs.

**Review by Abacn — APPROVED** [review]
LGTM

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python PreCommit

**Comment by ahmedabu98** [comment]
So looks like TestDirectRunner uses either `.wait_until_finish()` for batch pipelines or `on_success_matcher()` for streaming pipelines [when it runs](https://github.com/apache/beam/blob/8e217ea0d1f383ef5033ef507b14d01edf9c67e6/sdks/python/apache_beam/runners/direct/test_direct_runner.py#L45-L50). Otherwise, the `try` block is a noop and the pipeline is canceled in the following `finally` block.

In other words, it is by design that `TestDirectRunner` needs `on_streaming_matcher()` in pipeline options for streaming pipelines.

**Comment by johnjcasey** [comment]
Got it, that tracks

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
Run Python PreCommit

**Comment by Abacn** [comment]
FYI we can temporarily comment out test suites in PostCommit to get results more quickly and avoid burning out quota: https://github.com/apache/beam/pull/23480/commits/a3158eacc230328dcecd3286feeafe54aae268ed

**Comment by ahmedabu98** [comment]
Thanks @Abacn

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.8 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.9 PostCommit

**Comment by ahmedabu98** [comment]
Run Python 3.7 PostCommit

**Comment by ahmedabu98** [comment]
Run Python PreCommit
